File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int i = 0;
        int maxProfit = 0;
        for (int j = 0; j < prices.size(); j++) {
            if (prices[j] > prices[i]) {
                maxProfit = max(maxProfit, prices[j] - prices[i]);
            } else {
                i = j;
            }
        }
        return maxProfit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let maxprofit = 0;
    let i = 0
    for (let j = 1; j < prices.length; j++) {
        if (prices[j] > prices[i]) {
            maxprofit = Math.max(maxprofit, prices[j] - prices[i])
        } else {
            i = j
        }

    }
    return maxprofit

};